als-server/shared/src/test/scala/org/mulesoft/als/server/modules/project/ProfileConfigurationChangeListenerTest.scala

```diff
@@ -0,0 +1,128 @@
+    extends LanguageServerBaseTest
+    with ChangesWorkspaceConfiguration
+    with FileAssertionTest {
+
+  override implicit val executionContext: ExecutionContext =
+    ExecutionContext.Implicits.global
+
+  override val initializeParams: AlsInitializeParams = AlsInitializeParams(
+    Some(AlsClientCapabilities(serialization = Some(SerializationClientCapabilities(true)))),
+    Some(TraceKind.Off))
+
+  override def rootPath: String = "project"
+
+  val workspacePath: String = filePath(platform.encodeURI("project"))
+  val mainFileName: String  = "project/api.raml"
+  val profileUri: String    = filePath(platform.encodeURI("project/profile.yaml"))
+  val profile2Uri: String   = filePath(platform.encodeURI("project/profile2.yaml"))
+  val profile3: String      = filePath(platform.encodeURI("project/profile3.yaml"))
+
+  def buildInitParams(workspacePath: String): AlsInitializeParams =
+    buildInitParams(Some(workspacePath))
+
+  def buildInitParams(workspacePath: Option[String] = None): AlsInitializeParams =
+    AlsInitializeParams(
+      Some(
+        AlsClientCapabilities(customValidations = Some(CustomValidationClientCapabilities(true)),
+                              serialization = Some(SerializationClientCapabilities(true)))),
+      Some(TraceKind.Off),
+      rootUri = workspacePath,
+      hotReload = Some(true)
+    )
+
+  def buildServer(serializationProps: SerializationProps[StringWriter],
+                  notifier: ClientNotifier = new MockDiagnosticClientNotifier,
+                  withDiagnostics: Boolean = false): LanguageServer = {
+    val factoryBuilder: WorkspaceManagerFactoryBuilder =
+      new WorkspaceManagerFactoryBuilder(notifier, logger, EditorConfiguration())
+    val dm = factoryBuilder.buildDiagnosticManagers(Some(DummyProfileValidator))
+    val serializationManager: SerializationManager[StringWriter] =
+      factoryBuilder.serializationManager(serializationProps)
+    val pc = factoryBuilder.profileNotificationConfigurationListener(serializationProps)
+
+    val factory: WorkspaceManagerFactory = factoryBuilder.buildWorkspaceManagerFactory()
+
+    val builder =
+      new LanguageServerBuilder(factory.documentManager,
+                                factory.workspaceManager,
+                                factory.configurationManager,
+                                factory.resolutionTaskManager)
+    builder.addInitializableModule(serializationManager)
+    if (withDiagnostics) dm.foreach(m => builder.addInitializableModule(m))
+    builder.addRequestModule(serializationManager)
+    builder.build()
+  }
+
+  def buildSerializationProps(alsClient: AlsClientNotifier[StringWriter]) =
+    new SerializationProps[StringWriter](alsClient) {
+      override def newDocBuilder(prettyPrint: Boolean): DocBuilder[StringWriter] =
+        JsonOutputBuilder(prettyPrint)
+    }
+
+  def getDiagnostics(implicit diagnosticNotifier: MockDiagnosticClientNotifier): Future[PublishDiagnosticsParams] = {
```

Comment:
when do you need to test diagnostic in this suite?

Reply:
Nowhere, it was left here by mistake.